stglib/sig/cdf2nc.py

```diff
@@ -313,19 +370,25 @@ def ds_rename_sig(ds, waves=False):
 
 def fix_encoding(ds):
     """ensure we don't set dtypes uint for CF compliance"""
-    if "sample" not in ds.dims:
+    if "units" in ds["time"].encoding:
+        ds["time"].encoding.pop("units")
+
+    tstep = ds["time"][1] - ds["time"][0]  # use time step to select time encoding
+    # if "sample" not in ds.dims:
+    if tstep < np.timedelta64(1, "m"):
         print(
-            "make time encoding to dtype double because no sample dimension, round to milliseconds first"
+            f"make time encoding to dtype double because tstep {tstep} seconds is < 1 minute, round to milliseconds first"
```

Comment:
Rather than printing this to the screen, please add it to the netCDF history using utils.insert_history (that will also print it to the screen).

Reply:
changed as suggested.